Libraries/LibWeb/Painting/DisplayList.cpp

```diff
@@ -18,9 +18,20 @@ void DisplayList::append(Command&& command, Optional<i32> scroll_frame_id, RefPt
 String DisplayList::dump() const
 {
     StringBuilder builder;
-    for (auto const& command : m_commands) {
-        command.command.visit([&builder](auto const& cmd) { cmd.dump(builder); });
-        builder.appendff("\n");
+    size_t indentation = 0;
+    for (auto const& command_list_item : m_commands) {
+        auto const& command = command_list_item.command;
+        if (indentation > 0 && (command.has<PopStackingContext>() || command.has<Restore>()))
+            --indentation;
+
+        if (indentation > 0)
+            builder.append(MUST(String::repeated("  "_string, indentation)));
+        command.visit([&builder](auto const& cmd) { cmd.dump(builder); });
+        builder.append('\n');
+
+        if (command.has<ApplyCompositeAndBlendingOperator>() || command.has<ApplyFilter>() || command.has<ApplyOpacity>()
```

Comment:
maybe we could add some helper that returns bool indicating whether command increases nesting level and reuse it in DisplayListRecorder?

Reply:
I've added a field to all relevant commands. Ideally we could use that same field for `m_save_nesting_level` but that required more messing with `APPEND(..)` than I was comfortable with.

Reply:
@kalenikaliaksandr I've also found a way to support this in `APPEND(..)`, added it as a new commit